content/posts/jpa_spring_data_query_optimisations.en.md

```diff
@@ -0,0 +1,213 @@
+---
+title: "Tips & tricks for optimizing JPA queries"
+date: 2024-03-24T06:00:43+01:00
+draft: true
+  
+featuredImagePreview: "/assets/images/2024/03/tobias-fischer-PkbZahEG2Ng-unsplash.webp"
+featuredImage: "/assets/images/2024/03/tobias-fischer-PkbZahEG2Ng-unsplash.webp"
+images: ["/assets/images/2024/03/tobias-fischer-PkbZahEG2Ng-unsplash.webp"]
+tags:
+  - Java
+  - JPA
+  - Spring_Data
+
+---
+
+{{< style "text-align:center;" >}}
+_Picture of [Tobias Fischer](https://unsplash.com/fr/@tofi?utm_content=creditCopyText&utm_medium=referral&utm_source=unsplash)_
+
+{{< /style >}}
+
+Most of the Java developers I know don't really know what is under the hood of [Spring Data](https://spring.io/projects/spring-data) and [Java Persistence API (JPA)](https://docs.oracle.com/javaee/7/tutorial/persistence-intro.htm).
+In my opinion, it's mainly due to all the features provided by these specifications and frameworks. 
+They bring a lot of simplicity which make you forget SQL queries syntax.
+
+Unfortunately, when your stored data is coming to grow, querying against your database could be difficult. 
+The different queries you can indeed make your Java applications being slow and potentially break your SLOs.
+
+In this article, I tried to write down a bunch of tips & tricks to tackle this issue.
+Even if some are related to [Spring Data](https://spring.io/projects/spring-data), I think you can use most of them if you use a JPA in a standard way.
+
+You will see that even if we can consider using JPA easy at first glance, it can bring a lot of complexity.
+
+## Observe your application
+### Observe your persistence layer
+
+First and foremost, you **MUST** trace and monitor your persistence layer.
+
+If you use Hibernate (without Spring, Quarkus,...), you can get insights configuring the logger:
+
+```xml
+<logger name="org.hibernate.SQL">
+     <level value="debug"/>
+</logger>
+```
+If you use Spring (and JPA, Hibernate), you can also get them adding these configuration properties:
```

Comment:
I don't know about Spring, but for JPA `show_sql` is deprecated, see https://thorben-janssen.com/hibernate-logging-guide/#dont-use-showsql-to-log-sql-queries